thunder/tests/test_core.py

```diff
@@ -2313,6 +2313,42 @@ def forward(self, x):
     assert "t_fc2_weight" in sig.parameters
 
 
+@requiresCUDA
+def test_clone():
+    def foo(a):
+        return a.clone()
+
+    jfoo = thunder.jit(foo)
+    for shp in ((3, 5), [7], (8, 6, 4)):
+        for dev in (torch.device("cpu"), torch.device("cuda:0")):
+            for dt in (torch.float32, torch.float16, torch.bfloat16):
+                # there are issues with layouts other than strided; see
+                # test_clone_sparse_coo.
+                lout = torch.strided
+                b = jfoo(torch.randn(shp, device=dev, layout=lout, dtype=dt))
+                assert b.dtype == dt
+                assert b.layout == lout
+                assert b.device == dev
+                assert b.shape == torch.Size(shp)
+
+
+@pytest.mark.xfail(reason="randn() call fails, before getting to thunder")
```

Comment:
I don't think that thunder supports sparse tensors, but to create one you can call `to_sparse` on a non-sparse tensor

see https://pytorch.org/docs/stable/sparse.html

Reply:
I fiddled around with this a bit but hit another missing torch feature; it seems `repeat` uses `expand` under the hood, and `expand` doesn't work on sparse tensors.

I get the impression that dense/sparse layouts are not in wide use, and I don't think having our support go wider than torch's is valuable at this time. I'll leave the test just to document our shortcoming but ... let's move on.